### ISSUE TITLE
using tags to associate annotations

### ISSUE THREAD

**Issue by eltonteb** [pelagios/recogito2#299]
I'd like to try out "associating" place annotations with each other to identify and start to make a topology of place relations. I can do this using tags "en bloc" highlighting, but would there be a way of connecting two annotations without highlighting the whole text? Take the following example:
"On the Greek mainland facing the Cyclades Islands and the Aegean Sea the Sunium promontory..."
I'd want to connect "the Greek mainland" with "the Sunium promontory", and *then separately* "the Cycladic islands" and "the Aegean Sea" with "the Sunium promontory". Is there a way of tagging discrete elements rather than the whole clause?

**Comment by rsimon** [comment]
Since you're interested in relations between annotations, tags will, conceptually, not work. See this thread  on the Recogito Users forum, which has already a good discussion on various challenges involved:

http://commons.pelagios.org/groups/recogito-users/forum/topic/connected-annotations-and-multiple-possibilities/

**Comment by eltonteb** [comment]
of course Chiara has got there before me! What happened to the idea of the linking tool? (That is what I was getting at above.) It would be a start to identify that a relation exists. Of course, then we'd need a taxonomy to describe the relationship, but that's research. (We made a start in Hestia, based around the ideas of movement and transformation.)

**Comment by rsimon** [comment]
I'm happy to repurpose this issue for the "linking tool". However, for actual implementation in the forseeable future we're simply lacking the resources with the current grant. (Happy to hear ideas & volunteering though ;-) )

**Comment by eltonteb** [comment]
anyway, I've had a go at "tagging" relations in the first two paragraphs of my text, just so that you/we have a trial scenario. As you'll see, it's, well, complex... :)

What strikes me always when annotating, especially when thinking about relations, is the emphasis on interpretation at *every* turn. Because of that, I totally agree with the thrust of the commons thread that we would need to make any annotation of relations basic: the job will be to interpret the aggregated results.

**Comment by eltonteb** [comment]
The people at Lund/Umea are interested, as are - it seems - Chiara and the Leipzig posse. I'd be happy to try to get something going. Technically, what's the best way to think about it? As a plugin?

**Comment by rsimon** [comment]
I think the first step should simply be to hack a proof of concept into a fork of Recogito. This, inevitably, touches on various corners of the code base. Some code for dragging lines would be quick to hack. But that will literally only be the tip of the iceberg. I'd expect there will be __significant__ UI work involved. The backend work, for storing the relations, is probably much less an issue.

In any case: before knowing whether this can be a plugin, we need to think all the (modelling and implementation) aspects through to the end. And IMO there's so many aspects to consider (and parties to involve), we can't think the whole thing through without building a proof-of-concept and then iterating on that. So I'd say: fork & try!

**Comment by efi** [comment]
Couldn't the js frontend code of http://brat.nlplab.org/ be used do the heavy lifting UI-wise? I mean this could be sold as "expert mode" with a little different look & feel (not only highlights but all annotations "gabable" above the text)

**Comment by rsimon** [comment]
@efi: could be a good starting point for Lund/Umea! @eltonteb?

**Comment by eltonteb** [comment]
Could be. What does @ChiaraPalladino think? I'll also share this thread with the Lund/Umea people.

Of course, while I'm a willing volunteer to test drive applications, I'm not going to be able to do the mechanicing myself, so I would appreciate guidance how to proceed.

**Comment by ChiaraPalladino** [comment]
there are quite a lot of tools available to facilitate linking between annotations. i never tested brat, but the Perseids people used hypothes.is to establish connections between named entities directly on a webpage. something more refined should be provided by CWRC-Writer, which at the moment is only available as a test environment. 

my concern is about the semantic - let's call it, stability, of the connections established. as Elton says, to reduce the amount of interpretation should be our primarily concern. At least a rough semantic model of relations should be established as a basic vocabulary, by consistently explaining what is implied by each expression of relation. otherwise, free annotation may not be the best idea - it could end up in a bunch of annotated connections that nobody really knows how to use (because we want our material to be useful to other human beings, right?). 

right now, I would be in every sort of test that you may want to try for Recogito, but I maintain the honest opinion that some basic linguistic analysis should be a priority. E.g. treebanking geographical descriptions as they are found in the text and doing text mining from there, or using other systems to try to "encode" what is in the source. 

Another idea that I am currently testing is to annotate geospatial patterns with regular expressions. Regular expressions could be very useful to individuate and annotate recurrent expressive patterns in texts, supposing that geographical descriptions are somehow stable "encodings" of defined relations between entities/categories.

**Comment by rsimon** [comment]
Can you point to an example of they used hyptothes.is in this regard? Might be interesting to emulate their patterns in terms of UI and practices.

**Comment by ChiaraPalladino** [comment]
here is an explanation of the project: http://perseids.org/sites/joth/#index 
basically they used hypothes.is to annotate relations between personal names following the semantic standards of SNAP (if i remember correctly), then extracted the annotations for a visualization in GapVis.

**Comment by rsimon** [comment]
Is this also from the same activity https://github.com/perseids-project/perseids_docs/wiki/hypothes.is-annotations?

**Comment by ChiaraPalladino** [comment]
exactly.

**Comment by rsimon** [comment]
Ok, but then they were also "tagging", in a conventional sense, not explicitly building links _between annotations_ - at least not in the UI directly. They had, if I understand correctly, agreed on a common vocabulary beforehand, and then worked "Recogito style", as it were? 

The project description mentions a "transformation module". Do you know how this module transformed the annotations into networks? Perhaps this is something we can leverage, too. Especially since Recogito outputs Open Annotation already, and provides unique, dereferancable URIs for annotations.

**Comment by rsimon** [comment]
@eltonteb can you also take a look at the above links (http://perseids.org/sites/joth/#index, https://github.com/perseids-project/perseids_docs/wiki/hypothes.is-annotations)?

**Comment by ChiaraPalladino** [comment]
yes, they had a shared semantic vocabulary of relations derived from a simplified ontology, and used it to establish relations. 
i am not sure how the transformation module was, but the output was a graphic representation of personal relations in GapVis, you can see it here (the section "network of people"): http://perseids.org/sites/joth/#book/urn:cts:pdlrefwk:viaf88890045.003.perseus-eng1

**Comment by rsimon** [comment]
@eltonteb to return to your inital example - would the same approach work for that? (In which case the UI features in Recogito would suffice. But re-use of the Perseids transformation module could be of interest.)

**Comment by eltonteb** [comment]
I'll take a look tomorrow. @ChiaraPalladino, would you be around on Skype tomorrow, in case I need to ask you how things work?

**Comment by ChiaraPalladino** [comment]
i'll presumably be online, but the wifi in the office is down since this morning. however I wasn't involved in the project, so i know as much as there is in the github page.

**Comment by eltonteb** [comment]
I've had chance to look and play a little. Working with hypothesis.is isn't that different from Recogito, interestingly, which is a good thing I think (in terms of working within a visual standard). I'm just a little confused how one marks "relationships" in hypothesis, unless it's the same as I've been doing in Recogito - i.e. using a tag called "relation" and then using a typology of said relations. 

Is this what you meant by "the UI features in Recogito" @rsimon?

I like the linking tool in http://brat.nlplab.org/, but I'm fine with using tags, if that makes things easier.

Does this make sense, or have I got the wrong end of the stick as usual?

**Comment by rsimon** [comment]
<Is this what you meant by "the UI features in Recogito" @rsimon?>

What I meant was exactly that (I think) hypothes.is and Recogito work pretty much the same way, UI-wise. So whatever Perseids has done in terms of marking up relations in hypothes.is should work just as well in Recogito. My question to you, @eltonteb, is whether this way of marking up things is sufficient to what you need?

Did you get the linking to work in brat? I could view annotated texts, but somehow couldn't figure out how to create annotations (and links between them) myself.

**Comment by eltonteb** [comment]
well, since that's what I was already doing in Recogito, I guess it's ok :). Just one point though: I wasn't clear from the Perseids example how they disambiguated which concepts were being related. I get that the tag "relation" identifies that the entity has a relation, but with what? How, in other words, did they create their networks? Any ideas?

I confess that I didn't get beyond the tutorial with brat, which is probably telling in itself. The general interface reminds me of treebanking, which, while being logical and detailed, is far more complex than I think would be good for us to try to handle.

**Comment by efi** [comment]
@rsimon Did you try out the demo installation? http://weaver.nlplab.org/~brat/demo/latest - Did you log in with the demo credentials? _user_:"crunchy" _password_:"frog"

Just a general comment: I just pointed towards brat to hint at the UI tha talready allowed for "arrows" between annotations. The system with its backend may or may not be suitable for a certain application scenario. One can use only the js (visualisation/interaction) frontend code of brat. One can use brat after a data export. From CLARIN emerges WebAnno (https://webanno.github.io/webanno/) - a complete service-anabled annotation tool building on the brat UI. So the options are plenty (I already once wrote a half-baked API-compatible replacement for the original brat backend in ruby, so I could use the whole UI code, including collection manager, etc.) - the thing is one should choose the tool according to your needs.

**Comment by rsimon** [comment]
> "relation" identifies that the entity has a relation, but with what?

That's what I was wondering, too (and which was also among the points in the thread on Commons). I'm suspecting that the links are either somehow "inferred" (based on tagging conventions) and then the transformation module produces the network; or Perseids is taking an altogether different approach to what you're after @eltonteb. Either way, is it worth bringing someone in from the Perseids team? Perhaps through the Commons forum (rather than GitHub), as it's probably visible to the larger community?

**Comment by rsimon** [comment]
Thanks @efi, right - after logging in, things worked :-)

I second the point of choosing tools according to needs! And to look for workflows that combine different tools, rather than aiming to pack too many features into one. (Nonetheless my gut feeling is that an "arrow" tool may make sense for Recogit in the long run... in which case the JS from brat will definitely be helpful!) I'll need to look into WebAnno as well, thanks!

**Comment by eltonteb** [comment]
Done. I'd like to second a "pointy" tool to indicate relations, but, if we can find a way of getting tagging to work, that's fine by me for the time being.

**Comment by xarvelius** [comment]
![pausanias](https://cloud.githubusercontent.com/assets/11397961/20474457/f176634e-afc6-11e6-99ff-2894471d2065.png)
 Johan Åhfelt, Johan VB  and I have been trying to annotate Pausanias- the text is incredibly dense so we have made a scheme that we hope it would make sense. See diagram and here comes the analysis... 

We annotate: 

1. Entities (defined manually as we go along with recogito for places)
•	Persons (men, women, individual humans, gods, historical, fictional)
•	Places (buildings/infrastructures/natural environments)
•	Organizations (collectives of people, political, geographical, cultural)


2. Events (defined as contexts between people and organizations- manually as we go along we annotate the whole text, and then give it a category of event. We will have a platform to annotate but not sure it will be recogito. There is an event option in recogito but only has events and options add comments add tags). The text is very complex so there needs to be a different data model. More complex or more sensitive to identify for the cultural dimensions. Recogito is not made for cultural analysis. 
•	Wars/battles/blockades/ conflicts
•	Treaties 
•	Rituals
•	Deaths
•	Births (?!)

3. Relationships (between persons, organizations, places,- all the entities, discuss the analytical framework into a data model but this is not quite there yet but will define the limits of what we can do with analysis. Should be more simple than reality)
•	Friendship
•	 Alliances
•	Families

4. Attributes (each individual properties of an entity, one at a time)  

•	Materials
•	Infrastructural
•	Physical
•	Cultural properties for each entity at a time



Layer of visualizations beyond the spatial
Time (tagged in relation to global attributes) 

a.	present 
b.	past (historical or mythical

Weather (is there?)

**Comment by ChiaraPalladino** [comment]
pushing this up and tagging @mromanello here as a follow up. It would be interesting to study an integration of Brat in order to ensure connected annotations. Brat alone is usable but slightly (sic!) cumbersome. I don't know how much this would require, but it seems to me that many of us are interested in having this kind of feature somehow available in Recogito.

**Comment by rsimon** [comment]
An integration with Brat on the UI level is probably (very) difficult, as it works in a very different way. (From what I saw, it's all SVG in Brat. In other words, the whole document & text is essentially rendered as one big graphic. I'm worried this won't work efficiently on large texts - I mean, Elton-sized ;-) - and lots of annotations. But even if, it would mean an almost full re-implementation of either Recogito's or Brat's UI to make it work seamlessly.)

Is there perhaps a way to integrate on a workflow level? I.e. start working in one tool, transferring data to the other etc.?

Alternatively, I've been thinking about "native" ways to do this in Recogito. I could imagine a sort of extra dropdown option in the toolbar ("Annotation Mode: Normal | Relation"). Other than that, the whole process would stay the same. Select a text range, get the same popup (or do we want a slightly modified one?) But instead of being rendered as a text highlight, this type of annotation would get displayed as an arrow. The neat thing is that, because it would (internally) work like a normal annotation, we can re-use most of what's there. And, tech-wise, the arrow rendering could be done in CSS (i.e. just a slightly more clever graphical style for the annotation highlight), which means we know we'll be able to handle the same amounts of annotations/arrows as we do now.

**Comment by rsimon** [comment]
P.S.: the only thing that will require some internal tweaking is the fact that, obviously, the arrow shouldn't just enclose text, but link between "enclosed" annotations. I still think that we may want to treat this, internally, as an inferred connection rather than forcing this into the model too soon. It would be a bigger change and I'm suspecting we'll discover the subtleties of linking annotations once we've been doing it for a while - which, we can still make the model more explicit in this regard.

**Comment by ChiaraPalladino** [comment]
I agree with you. Brat is terrible on long documents, it would force to split them into multiple files. And the annotations themselves are, I suspect, quite heavy. The only cool thing about Brat is, I think, that it allows you to use and then retrieve an internal vocabulary from your annotations, i.e. you can assign your own vocabulary to the connections between annotations (for instance, classifying types of relations between places) and then export a list of those relations only. That creates a sort of internal vocabulary from your annotations, which can then be used for further annotation projects. 

I imagined it would have been painstaking to integrate the two tools. Personally, I think it would be nice to have this feature native on Recogito but we have to study it well. First, it should be conceived for bottom-up vocabularies, because there is no canonical reference whatsoever here. You can choose one from an external reference, but there is nothing like a gazzetteer that we can use. Therefore, it has to be very flexible. 
In terms of visualization, it's fine to have arrows or whatever else looks nice to visualize the connections on the interface - my worry is about the export formats and how to use those annotations. We want to 1. mark these relations, 2. classify them semantically (by assigning values to each connection, e.g. partOf, sonOf, closeTo...), 3. be able to link multiple entities together, i.e. one single entity does not necessarily have just one relation, it may have multiple connections at multiple hierarchical levels, 4. export these annotations in a way that makes sense (RDF?) for using them in other projects. 

We still don't know what to do with multi-level annotations here, which I think are underrated in their potential. But as I mentioned before, it may be too complex to use them for relations.

**Comment by mromanello** [comment]
Thanks @ChiaraPalladino for making me aware of this thread! What got me thinking of a possible integration between Brat and Recogito is the increasing support for named entities (i.e. not just place names) that it's being added to this tool.

I'm currently using Brat in several projects for annotations of named entities, and while I find its UI pretty neat, it lacks many features for managing large annotation projects -- something at which Recogito is pretty good (said by someone who doesn't know the ins and outs of the tool, though).

I really love the new feature that allows one to call an external NE tagging service from within Recogito, and I believe this opens up new great possibilities. 

In terms of UI and visualization @rsimon 's idea seems very sensible. What is essential from my point of view is the ability to give a label to each relation and eventually attached a value to it -- but I just don't know to what extent this fits (or not) into the current data model.

Concerning how to export the relation annotations, I'd probably go for a format as general as possible, then leaving the transformation into other formats (e.g. RDF) to the users, but again I'm not sure how this relates to the current export functionalities of Recogito.

Anyway, I'd be happy to follow Recogito's development in this direction, and to contribute to the discussion.

**Comment by rsimon** [comment]
Been thinking about the line drawing again, and potential Brat-like functionality for Recogito... Three specific questions that occurred to me:

- Would you expect/want to have the arrows on the screen all the time? Or would it work if the arrows/lines would only show when you click on a specific annotation? I wonder whether we'd end up in a hairball-like text if we showed __all__ the arrows all the time. (I openly admit that it would probably make implementation a lot easier, too, if we don't show the arrows all the time.) @mromanello I assume you have some examples of "real" Brat-annotated documents. Can you post a screenshot so I can get an impression of the "density" of relations you are working with?

- I'm beginning to think that visualization could be relatively straigthforward (if we don't show everyting all the time at least). But I'm also wondering how we would represent this in the model. I'm inclined to consider this truly an annotation that annotates two other annotations, i.e. in Open Annotation terminology an annotation that has two targets, each of which are annotations themselves. Thoughts?

- Obviously, we'll want to attach "bodies" (OA terminology) to this annotation too. In terms of the model, we could attach any body (since Recogito would treat it just like any other annotation in the system). But in the user interface, it's probably preferable to treat it in a simpler way. @ChiaraPalladino I stumbled across your annotated version of the Catalog of Ships, where you noted the commanders for the places/peoples. I assume this is exactly one of those cases you would have used a connection? So if we  allow, in the UI, adding tags to the link ("commander"), would that be enough (at least in the first instance)?

**Comment by ChiaraPalladino** [comment]
So, in order: 

- Not only it would be preferable, but highly desirable, to be able _not_ to show the arrows all the time. It would become a total chaos in a minute if we couldn't. What about a "show/do not show relations" button? 
- I don't see big limitations in doing it as a connection between annotations. In fact, that's what it is in the Recogito workflow. The only problem I see here is multiple types of relations, for which it could become quite heavy in terms of data export, but I'm not sure we wouldn't have this problem if we adopted any other method. Perhaps @efi has deeper thoughts than me on this? 
- The Catalog of Ships was a short project that I did for the students here in Tufts :) so the type of annotation that is in there is very elementary - but yeah, in the first instance that would be a easy type of connection that would come quite immediate to anyone annotating a text. Then we could adapt the model to more complex types of relations, e.g. annotating strings of texts as relations or multiple relations to the same target. 
- Directionality is not _that_ important in the majority of relations. Yes there are hierarchical relations like partOf, or in terms of people, sonOf and fatherOf, for example, but this hierarchy could be represented visually if we gave them the right semantic definition (at least in my opinion, but some could disagree on this). Maybe a simple visual option such as arrows vs simple lines could help in distinguishing directional relations and non directional relations? 
It's actually more important to have multiple annotations, i.e. to be able to establish a link to an entity more than once. I'll make a concrete example I stumble upon quite frequently: 

"The boundaries of France are Spain, Belgium, Luxembourg, Switzerland and Italy". 
In such sentence there are: 
- a semantically defined relation, "boundary". 
- "France" as annotated place 
- "Spain", "Belgium", "Luxembourg", "Switzerland" and "Italy" as annotated places. 
- relation: "France" - "Boundary" - "Spain" 
- relation: "France" - "Boundary" - "Belgium"
- relation: "France" - "Boundary" - "Luxembourg"
- relation: "France" - "Boundary" - "Switzerland"
- relation: "France" - "Boundary" - "Italy"
Total: 11 annotations. And that's the easiest case I could think of :)
Now it is not necessarily true that this relation implies a directionality, because the boundary is on both sides of France and the other country. However, since France is the common subject to all triples, we should be able to retrieve a geometry to represent this visually. Same said for linear relations, such as directions and distances.

**Comment by efi** [comment]
I'm a bit short on time right now, so just a few comments:

To model a relation as fully qualified annotation between two (or  more, but that's of course another topic) annotations (rather than "unannotated" text passages) ist the most sensible thing to do, because you may need to add other values to it + chances are high you want to assert some things about the "endpoints" of your relation, anyways.

Multiple relation types are absolutely necessary to express multiple semantics of relatedness - whether to do this on a system level or through "tags" or another mechanism on the relation-annotation level would not make a big difference for most use-cases, I think.

Then, whether relations should have a "direction" or rather a "role" attached to every endpoint (allowing for symmetrical relations where due) should be discussed in detail. Sometimes it is not _that_ important, sometimes it is crucial to give clues about the specific part an entity plays in the relation.

I guess I could start to contribute some more thoughts or even work on some prototypical code in a week or so.

**Comment by mromanello** [comment]
Hi all - 

just some thoughts on topics/issues that have emerged in the few last messages:

- @rsimon here an example of documents where the visualization of all relations may start to get a bit messy (especially with those span several lines). I agree @ChiaraPalladino's idea of offering a button to toggle on/off the relations, perhaps even something like "hide already checked relations" (although I don't know if it is planned to track the validation status of relations -- I, for one, would find it certainly useful).

![brat_example](https://cloud.githubusercontent.com/assets/310693/22645762/e8fcbba8-ec68-11e6-9a71-f5024a01e665.png)

- concerning the `body` of relations: I'd be willing to attach a (CTS) URN (or other URI) to each entity/relation, thus I'd be happy with any system that leaves as much freedom on this respect as possible.

- for comparison, this is a snippet from brat's config file for annotations (from my own data, actually the same annotation scheme displayed in the image above):
    ```
    [entities]


    AAUTHOR
    AWORK
    REFAUWORK
    REFSCOPE

    [relations]

    Scope    Arg1:REFAUWORK|AWORK|AAUTHOR, Arg2:REFSCOPE
    ```
    as you can see, the directionality is not explicitly set, but i's always assumed that the relations are directed (arg1 => arg2). What this is very good for -- from the point of view of linguistic annotations, at least -- is the ability to validate (and possibly flag) entity/relation annotations. For example, the type of `Arg2` in a `scope` relation must always be a `REFSCOPE`. (btw, is there already something similar in Recogito, for entities?).

**Comment by rsimon** [comment]
Hi all,

thanks, that's all great input.

So, in the most basic case, the annotation with two targets could work in principle? I.e. (referring to @ChiaraPalladino's example above): target 1 = "France", target 2 = "Spain". The relation ("Boundary") could simply be a single (tag-)body on the annotation. This also seems to work with @mromanello's BRAT examples, right? (Two text snippets would independently be tagged as "REFAUWORK" and "REFSCOPE", respectively; and then the connecting annotation would have a single "Scope" body). 

And so far, things are in line with what I was expecting. However, it seems we'll also need to provide room for a third target, in order to (optionally) point to a text snippet that expresses the  relation in the narrative, right? (I.e. going back to @ChiaraPalladino's example: "The boundaris of". Maybe it's not strictly necessary to mark this up in the basic example above. But I'm sure Chiara can come up with lots of additional examples ;-) Not sure how to best name this "arrow"... but in any case it feels this become a relevant feature, even if we try to keep things to a minimum, feature-wise.

**Comment by eltonteb** [comment]
Dear all,

Having managed to neglect / miss / ignore @rsimon's repeated invitations to contribute to this thread, here's my tuppenceworth. I'm very much indebted to @ChiaraPalladino (as always) for articulating the key principles.

This is the first line of a text I'm interested in: Pausanias's Description of Greece:

On the Greek mainland facing the Cyclades Islands and the Aegean Sea the Sunium promontory stands out from the Attic land.

Thinking of Chiara's Paris example, I reckon that the annotations I'd like to make are as follows:
- Sunium promontory - annotated place (subject)
- Greek mainland, Cyclades Islands, Aegean Sea, Attic land - annotated places [objects]
- relation: Sunium promontory - Greek mainland ["on"]
- relation: Sunium promontory - Cycladic islands ["facing"]
- relation: Sunium promontory - Aegean Sea ["facing"]
- relation: Sunium promontory - Attic land ["stands out from"]

Here's the thing. I'm less interested in the specific relational identifiers (those in the square brackets) as abstracting from them. There are various ways to do this undoubtedly, and Chiara has her own ideas and reason. For me, I'd like to apply the methodology that we developed in Hestia: does the relation describe movement and/or transformation. Taking the example above, all these items define (according to my schema) the same relation: geographical description or proximity, where there is no movement or transformation involved.

So, if Recogito allows me (a) to annotate the relations above, (b) include directionality, and (c) define/tag the relationship schema I'm using (in that not all relations are the same), then, yes, this is all fine.

@rsimon: re. your last comment, can you help me understand when one would want to use two or three targets? Does it relate to my point (c)?

**Comment by rsimon** [comment]
@eltonteb 

>  Does it relate to my point (c)?

I think it does. The difference is whether you just want to use a controlled vocabulary to "attach labels to your arrows" (either terms like "on", "faces", "stands out", etc. - or an abstraction from that: "description", "proximity", movement), __or__ whether you are also interested in highlighting those phrases in the text that spell out the relation.

In the former case, you'd just highlight "Sunium promontory" and "Aegean Sea", and then drag a line labeled "facing" between them (or "description", or "proximity" - or all of the above). In the latter case, you'd want to highlight "Sunium promontory", "Agean Sea" and "facing" in the text. Then you'd drag a line between the Sunium and Aegean Sea highlights (for the relation) and label that line with your vocab, just like before. But __in addition__, you'd have some sort of additional line (perhaps automatically labeled as "context" or whatever) pointing to "facing".

My impression is that you're really only interested in the simple case though, i.e. two target + your own controlled vocab?

**Comment by rsimon** [comment]
> Here's the thing. I'm less interested in the specific relational identifiers (those in the square brackets) as abstracting from them. 

P.S.: I think in terms of the process, whether you'd use a vocab derived from the text or a more narrow abstraction (or both) wouldn't make a difference from the technical point of view. In Recogito, it's just tagging after all.

**Comment by eltonteb** [comment]
@rsimon: ah, I see. Yes, I had in mind scenario 1 (two tags), but I can also see the virtue of scenario 2 (three tags), where it might be useful to identify/capture the precise term being used in the text to define the relation (before my interpretation and gloss in the form of controlled vocabulary). That won't make sense for every relation though, and probably only when I would be using the original language text. Still, it would be good to have the option.

PS. That was my impression too, in which case Recogito is just fine for what I want to be able to do :)

**Comment by xarvelius** [comment]
I think "description", "proximity", “movement” is adequate if the overarching question is human movement

since Pausanias is so complex we have produced a rather complex scheme here at Humlab (but I think for the sake of Humanistic inquiry we need to compress it, but I am still working on it. I will get back to you in about two weeks as planned!

Anna

[cid:63D4A0E7-616E-486D-A2B8-B5FA06B3F10C@.]



Dr. Anna Foka
Associate Professor
Research Coordinator
Humlab
www.annafoka.com<http://www.annafoka.com>

On 13 Feb 2017, at 12:30, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:

description", "proximity", movement

**Comment by xarvelius** [comment]
Another issue I see is not being able to tag over what eg elton tagged. Not on the same (Elton established) metadata tags. And there are some that, as I go through the document, I may see in a different light. I can, however, erase Elton’s tags if I want to. May it’s a bug in the system?

An event could also be mapped as time. Then I guess we can gather all time and add a period or estimate. I suppose these should be canonized so they are (I guess) better machine identifiable. Most dates are, in any case, either uknown (for example: once) or imprecise (the time of Antigonus, or the time of the Peloponnesian War: in the latter case this could be standardized eg PL War or P (=Persian) War).

Semantic annotation is good with Jet Lag (or not!)

I am enjoying more and more the recogito division of places, people, and events. I do think they require further clarification but tagging makes things easier.

More anon,

Anna



Dr. Anna Foka
Associate Professor
Research Coordinator
Humlab
www.annafoka.com<http://www.annafoka.com>

On 13 Feb 2017, at 03:36, eltonteb <notifications@github.com<mailto:notifications@github.com>> wrote:


@rsimon<https://github.com/rsimon>: ah, I see. Yes, I had in mind scenario 1 (two tags), but I can also see the virtue of scenario 2 (three tags), where it might be useful to identify/capture the precise term being used in the text to define the relation (before my interpretation and gloss in the form of controlled vocabulary). That won't make sense for every relation though, and probably only when I would be using the original language text. Still, it would be good to have the option.

PS. That was my impression too, in which case Recogito is just fine for what I want to be able to do :)

**Comment by rsimon** [comment]
Hi Anna,

you can add additional tags (not just delete existing ones). Did you hit Enter after typing the tag?

**Comment by xarvelius** [comment]
But of course. It doesn't always work on google chrome/ macos (I kept trying for sometime) but could be that I am at a hotel and the connection can be shaky at times.



Sent from ?????, my iPhone

On 15 Feb 2017, at 22:57, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Anna,

you can add additional tags (not just delete existing ones). Did you hit Enter after typing the tag?

-

**Comment by rsimon** [comment]
Can anyone (preferably) with a Mac try to reproduce? (I tried, it worked everytime - Chrome/Linux, Chrome/Windows). Re shaky connection: does the header bar show a warning (it should normally show a GoogleDocs-like "Saving..." -> "Annotation Saved" indication).

**Comment by rsimon** [comment]
P.S.: Can you explain the exact steps you are doing? It's difficult to guess otherwise.

**Comment by xarvelius** [comment]
Okay!
So I click on the tag that Elton has already placed. I then click the annotation window area next to existing tags that has the 'add tag' option and I add a word there e.g. For the phrase 'Apollo Patroos', I want to tag the word 'epithet' on Patroos for example. Then I type in the word, And then I press 'OK and next' as a button option. Nothing happens. No pop ups, nothing. The existing annotation window closes and when I click on the annotated word/ phrase again resurfaces without my newly added annotation on.

Does this make sense?



Sent from ?????, my iPhone

On 15 Feb 2017, at 23:33, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:


P.S.: Can you explain the exact steps you are doing? It's difficult to guess otherwise.

-

**Comment by rsimon** [comment]
You currently need to hit 'Enter' after typing the tag ('epithet'). Then click OK.

**Comment by xarvelius** [comment]
OK! Good to know!

Sent from ?????, my iPhone

On 16 Feb 2017, at 00:25, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:


You currently need to hit 'Enter' after typing the tag ('epithet'). Then click OK.

-

**Comment by ChiaraPalladino** [comment]
> And so far, things are in line with what I was expecting. However, it seems we'll also need to provide room for a third target, in order to (optionally) point to a text snippet that expresses the relation in the narrative, right? (I.e. going back to @ChiaraPalladino's example: "The boundaries of". 

I am in a hurry now (as you may know :D ), but I'm just noting that I _really_ value this aspect. I think it is crucial for Classical texts, because we simply don't have any canonical reference for spatial linguistic expressions for those, and this is the only way I can see that we can start to gather some use cases in a meaningful way. My interest here is pretty linguistic, not so much about abstracting a spatial model, which is Elton's case. Both approaches are very important at this stage of research, so I think we should think of a way to combinate the two "tags". 

Ps. I work on Mac and everything concerning tags is fine.

**Comment by rsimon** [comment]
Great! Then I would - tentatively and cautiously - say we have a plan ahead :-)

**Comment by xarvelius** [comment]
Hello all- I will get some Stanford user comments to you guys as soon as I get to my office.

The 'enter' button as an essential part of the process of tagging did not come naturally for many users. Generally speaking, however, they found the process, and especially the interface, quite amazing.

And how could they not!

Sent from ?????, my iPhone

On 17 Feb 2017, at 02:47, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:


Great! Then I would - tentatively and cautiously - say we have a plan ahead :-)

-

**Comment by rsimon** [comment]
Hi @xarvelius,

great. Yes, I also saw other users have difficulty with Enter being required. (It seemed to me to depend a bit on whether the user is trying to add a single tag, or multiple, the first time they are doing it. When adding multiple, the act of adding them one by one, hitting Enter in between, seems to come more naturally.) But in any case: tags that were typed should be added to the annotation in any case when OK is clicked. I created an issue for this: https://github.com/pelagios/recogito2/issues/364

**Comment by xarvelius** [comment]
Most people these days are familiar with social media interfaces where you tag using # and ‘tag’

A
Dr. Anna Foka
Associate Professor
Research Coordinator
Humlab
www.annafoka.com<http://www.annafoka.com>

On 18 Feb 2017, at 00:52, Rainer Simon <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @xarvelius<https://github.com/xarvelius>,

great. Yes, I also saw other users have difficulty with Enter being required. (It seemed to me to depend a bit on whether the user is trying to add a single tag, or multiple, the first time they are doing it. When adding multiple, the act of adding them one by one, hitting Enter in between, seems to come more naturally.) But in any case: tags that were typed should be added to the annotation in any case when OK is clicked. I created an issue for this: #364<https://github.com/pelagios/recogito2/issues/364>
